File: C++/Question-3.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

void guessNumber() {
    srand(time(0));
    int number = rand() % 100 + 1;
    int guess;

    while (true) {
        std::cout << "Guess the number (between 1 and 100): ";
        std::cin >> guess;

        if (guess < number) {//changed the greater than(>)to less than(<)
            std::cout << "Too low! Try again." << std::endl;
        } else if (guess > number) {  //changed the less than equal to(<=)to greater than(>)
            std::cout << "Too high! Try again." << std::endl;
        } else {
            std::cout << "Congratulations! You guessed the number." << std::endl;
            break;
        }
    }
}

int main() {
    guessNumber();
    return 0;
}

File: Java/Question-3.java
import java.util.Scanner;
import java.util.Random;

public class NumberGuessingGame {
    public static void guessNumber() {
        Random random = new Random();
        int number = random.nextInt(100) + 1; // Generates a random number between 1 and 100
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Guess the number (between 1 and 100): ");
            int guess = scanner.nextInt();

            if (guess < number) {
                System.out.println("Too low! Try again.");
            } else if (guess > number) {
                System.out.println("Too high! Try again.");
            } else {
                System.out.println("Congratulations! You guessed the number.");
                break;
            }
        }
    }

    public static void main(String[] args) {
        guessNumber();
    }
}